sdk/storage/src/common/shared_key_policy.cpp

```diff
@@ -46,17 +46,23 @@ namespace Azure { namespace Storage {
 
     // canonicalized headers
     const std::string prefix = "x-ms-";
+    std::vector<std::pair<std::string, std::string>> ordered_kv;
     for (auto ite = headers.lower_bound(prefix);
          ite != headers.end() && ite->first.substr(0, prefix.length()) == prefix;
          ++ite)
     {
       std::string key = ite->first;
-      std::string value = ite->second;
       std::transform(key.begin(), key.end(), key.begin(), [](char c) {
         return static_cast<char>(std::tolower(static_cast<unsigned char>(c)));
       });
-      string_to_sign += key + ":" + value + "\n";
+      ordered_kv.emplace_back(std::make_pair(std::move(key), ite->second));
     }
+    std::sort(ordered_kv.begin(), ordered_kv.end());
```

Comment:
I didn't know that the string to sign is alphabetical for each key, learnt something today :).